FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift

```diff
@@ -194,10 +197,61 @@ import Foundation
       guard await manager.checkNotificationForwarding() else {
         throw AuthErrorUtils.notificationNotForwardedError()
       }
-      return try await verifyClAndSendVerificationCode(toPhoneNumber: phoneNumber,
-                                                       retryOnInvalidAppCredential: true,
-                                                       multiFactorSession: multiFactorSession,
-                                                       uiDelegate: uiDelegate)
+
+      let recaptchaVerifier = AuthRecaptchaVerifier.shared(auth: auth)
+      try await recaptchaVerifier.retrieveRecaptchaConfig(forceRefresh: true)
+
+      switch recaptchaVerifier.enablementStatus(forProvider: .phone) {
+      case .off:
+        return try await verifyClAndSendVerificationCode(
+          toPhoneNumber: phoneNumber,
+          retryOnInvalidAppCredential: true,
+          multiFactorSession: multiFactorSession,
+          uiDelegate: uiDelegate
+        )
+      case .audit:
```

Comment:
Can these two cases be combined?

Reply:
Hmm, we can combine these, but should we do two comparisons or one for state?

Reply:
we should probably keep them seperate for now.  The Audit flow later will be updated when reCAPTCHA v2 deprecation happens. This will help the later refactor/deprecation work for AUDIT mode.

Reply:
The statement looks exactly the same - so how about `case .audit, .enforce:`?

Reply:
The `retryOnInvalidAppCredential` value is different for both of these 
https://github.com/firebase/firebase-ios-sdk/blob/phone-auth-login/FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift#L223